### ISSUE TITLE
Spamhaus incorrect operation

### ISSUE THREAD

**Issue by recoilstv** [mail-in-a-box/mailinabox#2288]
Hi! When checking the system status, I receive the following message: This domain is listed in the Spamhaus Domain Block List (code 127.255.255.254), which may prevent recipients from receiving your mail. See http://www.spamhaus.org/dbl/ and http://www.spamhaus.org/query/domain/mydomain. The same situation happens with my IP. But there are no problems when checking the status by the link. Because of this, I can't send and receive mail.

**Comment by Kirutian** [comment]
My box was doing the same thing, and it has suddenly stopped. How about yours?

**Comment by recoilstv** [comment]
Hi. I removed all the Spamhaus settings from the Postfix configuration and wrote on the product support forum

**Comment by blok82** [comment]
i had this a few times.
Spamhaus support will remove you from their list if its a false positive (which occurs too often imho). There is a form for that in the links you showed. Its done in a few hours, but still.... annoying

**Comment by recoilstv** [comment]
> i had this a few times. Spamhaus support will remove you from their list if its a false positive (which occurs too often imho). There is a form for that in the links you showed. Its done in a few hours, but still.... annoying

I wrote above that there are no problems when checking the status of the link on the Spamhaus website.

**Comment by saesrpg** [comment]
https://www.spamhaus.org/news/article/807/using-our-public-mirrors-check-your-return-codes-now.

https://www.spamhaus.com/resource-center/if-you-query-spamhaus-projects-dnsbls-via-ovhclouds-dns-move-to-the-free-data-query-service/

Seems to be affecting people on OVH

**Comment by florisre** [comment]
yes, this is a problem. My box is rejecting all emails as spam, since it can no longer properly get spamhaus info.

**Comment by kiekerjan** [comment]
This might happen when you're not using the Mail in a box built-in DNS resolver. Spamhaus [will return error codes 127.255.255.254 and 127.255.255.255](https://www.spamhaus.com/product/help-for-spamhaus-public-mirror-users/) for public/open resolvers, or when a user makes an excessive number of queries.
Did you change anything to the DNS settings of the box?

**Comment by florisre** [comment]
No, I did not change anything. However, the default set up had 1.1.1.1 as 
DNS server. Changing to 127.0.0.1 fixed the issue *temporarly*. To fully 
fix this, I got a DQS key from spamhaus and changed the postfix config to 
use it. Now everything is working as expected.

28.10.2023 21:35:10 KiekerJan ***@***.***>:

**Comment by kiekerjan** [comment]
> However, the default set up had 1.1.1.1 as DNS server.

That should not be the case. But perhaps your VPS provider overwrites the MiaB configuration? (it has happened before) How did you deploy this installation.
> Changing to 127.0.0.1 fixed the issue *temporarly*

Why temporarily?

**Comment by florisre** [comment]
Not sure why temporarily. The DNS configuration persists, I am assuming 
maybe too many request from my DNS provider's network?

29.10.2023 14:37:44 KiekerJan ***@***.***>:

**Comment by florisre** [comment]
> > However, the default set up had 1.1.1.1 as DNS server.
> 
> That should not be the case. But perhaps your VPS provider overwrites the MiaB configuration? (it has happened before) How did you deploy this installation.

I ran into the issue again. It does not seem to be my cloud provider overwriting the setting, but rather the setting not being applied successfully during setup. I was able to resolve this with creating `/etc/resolvconf/resolv.conf.d/head` and applying it with `resolvconf`. After restarting `resolved` everything worked as expected. Maybe this is about Ubuntu changing how DNS servers are configured, or maybe also about different networking setups used by VPS providers?

**Comment by kiekerjan** [comment]
> My box is rejecting all emails as spam, since it can no longer properly get spamhaus info

In the meantime I created [this](https://github.com/mail-in-a-box/mailinabox/pull/2332) so that at least the mail is not rejected as spam for this reason.
